Project/core/Payment.cs

```diff
@@ -6,64 +6,41 @@
 
 namespace Project
 {
+    [Serializable]
     class Payment
     {
-        private Client client;
-        private List<SinglePayment> singlePayments;
+        private DateTime date;
+        private string title;
+        private bool isPaid;
+        private double amount;
 
         //Konstruktory 
 
-        public Payment(Client client)
+        public Payment(string title)
         {
-            this.client = client;
-            this.singlePayments = new List<SinglePayment>();
+            this.date = DateTime.Now;
+            this.title = title;
+            this.isPaid = false;
+            this.amount = Amount();
         }
 
-        // Metody dodatkowe
-
-        public void AddSinglePayment(SinglePayment singlePayment)
-        {
-            singlePayments.Add(singlePayment);
-        }
+        //Getery i Setery
 
-        public void AddReserwation(Reservation reservation)
-        {
-            singlePayments.Add(new SinglePayment("Reserwation", reservation.Cost(), 1, false, reservation.CheckInDate));
-        }
+        public string Title { get => title; set => title = value; }
 
-        public void AddAdvance(Reservation reservation)
-        {
-            singlePayments.Add(new SinglePayment("Advance", reservation.Advance(), 1, false, DateTime.Now));
-        }
+        public bool IsPaid { get => isPaid; set => isPaid = value; }
 
-        public int GetCountOfNotPaiedSinglePayment()
-        {
-            return singlePayments.FindAll(sp => sp.IsPaid == false).Count();
-        }
+        // Metody dodatkowe
 
-        public bool AdvanceIsPaid()
+        public virtual double Amount()
```

Comment:
To powinna być metoda abstrakcyjna, czyli ze słowem kluczowym abstract i bez ciała metody